posthog/api/test/test_decide.py

```diff
@@ -239,6 +239,55 @@ def test_feature_flags(self):
             self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertEqual(response.json()["featureFlags"], ["default-flag"])
 
+    def test_feature_flags_v3_json(self):
+        self.team.app_urls = ["https://example.com"]
+        self.team.save()
+        self.client.logout()
+        Person.objects.create(team=self.team, distinct_ids=["example_id"], properties={"email": "tim@posthog.com"})
+        FeatureFlag.objects.create(
+            team=self.team, rollout_percentage=50, name="Beta feature", key="beta-feature", created_by=self.user
+        )
+        FeatureFlag.objects.create(
+            team=self.team,
+            filters={"groups": [{"properties": [], "rollout_percentage": None}]},
+            name="This is a feature flag with default params, no filters.",
+            key="default-flag",
+            created_by=self.user,
+        )  # Should be enabled for everyone
+        FeatureFlag.objects.create(
```

Comment:
a test with non-multivariate payload would be nice as well

Reply:
minor nit: Does it make sense to have `payloads` for variants in an outer key as well, based off of the key?

I kinda like this because it's consistent / we don't need to have 2 different codepaths for non-multivariate & multivariate flags / easier to send from the frontend in one place.

Reply:
Can end up with collisions if you have variant keys repeated across different flags. Would need atleast an extra layer of nesting for key->variant key as the identifier

Reply:
Not sure I understand why there will be collisions, since the flag payload is determined by the flag ID? 🤔 .

I meant something like:
```
flag = {
name: blah,
filters: {
    groups: {}
    multivariate: {}
    payloads: {
          'true': blah blah // boolean case
          // multivariate case
         'control': blah blah
        'test': blah 2
    }
},
```

Reply:
Oops, youre right

Reply:
Done